src/core/qgspathresolver.h

```diff
@@ -171,6 +172,114 @@ class CORE_EXPORT QgsPathResolver
     % End
 #endif
 
+
+
+    /**
+     * Sets a path writer function, which allows for manipulation of paths and data sources prior
+     * to writing them to the project file.
+     *
+     * The \a writer function must accept a single string argument (representing the original file path
+     * or data source), and return a processed version of this path.
+     *
+     * The path writer function is called before any bad layer handler.
+     *
+     * If multiple writers are set, they will be called in sequence based on the order in which
+     * they were originally set.
+     *
+     * \returns An auto-generated string uniquely identifying the writer, which can later be
+     * used to remove the writer (via a call to removePathWriter()).
+     *
+     * \see removePathWriter()
+     * \since QGIS 3.22
+     */
+#ifndef SIP_RUN
+    static QString setPathWriter( const std::function< QString( const QString &filename )> &writer );
+#else
+
+    /**
+     * Sets a path writer function, which allows for manipulation of paths and data sources prior
+     * to writing them to the project file.
+     *
+     * The \a writer function must accept a single string argument (representing the original file path
+     * or data source), and return a processed version of this path.
+     *
+     * The path writer function is called before any bad layer handler.
+     *
+     * If multiple writers are set, they will be called in sequence based on the order in which
+     * they were originally set.
+     *
+     * Example - replace path with a variable:
+     *
+     * \code{.py}
+     *   def my_processor(path):
+     *      return path.replace('c:/Users/ClintBarton/Documents/Projects', '@projectdir@')
```

Comment:
@manisandro this does not translate correctly in the pyQGIS documentation (https://qgis.org/pyqgis/master/core/QgsPathResolver.html#qgis.core.QgsPathResolver.setPathWriter)

Reply:
So I should use i.e. $ rather than @?

Reply:
no idea, sorry.